mat/diagonal_test.go

```diff
@@ -9,6 +9,7 @@ import (
 	"reflect"
 	"testing"
 
+	"golang.org/x/exp/rand"
```

Comment:
I think Dan prefers to have 3rd party imports in their own list, separated from the `gonum.org/...` ones.
(ie: insert a blank line b/w `exp/rand` and `blas/blas64`.)